aicsimageio/readers/bioformats_reader.py

```diff
@@ -0,0 +1,457 @@
+            mo = loci.formats.in_.DynamicMetadataOptions()
+            mo.set("nativend2.chunkmap", "False")
+            self._r.setMetadataOptions(mo)
+
+        self._nseries = self._r.getSeriesCount()
+        self.set_series(series)
+
+    def set_series(self, series=0):
+        self._r.setSeries(series)
+        self._core_meta = CoreMeta(
+            (
+                self._r.getSizeT(),
+                self._r.getEffectiveSizeC(),
+                self._r.getSizeZ(),
+                self._r.getSizeY(),
+                self._r.getSizeX(),
+                self._r.getRGBChannelCount(),
+            ),
+            _pixtype2dtype(self._r.getPixelType(), self._r.isLittleEndian()),
+            self._r.getSeriesCount(),
+            self._r.isRGB(),
+            self._r.isInterleaved(),
+            self._r.getDimensionOrder(),
+        )
+
+    @property
+    def core_meta(self) -> CoreMeta:
+        return self._core_meta
+
+    def open(self) -> None:
+        """Open file."""
+        self._r.setId(self._path)
+
+    def close(self) -> None:
+        """Close file."""
+        try:
+            self._r.close()
+        except (ImportError, RuntimeError):
+            pass
+
+    def to_numpy(self, series: Optional[int] = None) -> np.ndarray:
+        """Create numpy array for the current series."""
+        return np.asarray(self.to_dask(series))
+        # doesn't seem any faster ...
+        # from itertools import product
+        # nt, nc, nz, *_ = self._shape
+        # out = np.empty(self._shape)
+        # for z, c, t in product(range(nz), range(nc), range(nt)):
+        #     out[t, c, z] = self._get_plane(z, c, t)
+        # return out
+
+    def to_dask(self, series: Optional[int] = None) -> "da.Array":
+        """Create dask array for the current series."""
+        import dask.array as da
+
+        if series is not None:
+            self._r.setSeries(series)
+
+        nt, nc, nz, ny, nx, nrgb = self.core_meta.shape
+        chunks = ((1,) * nt, (1,) * nc, (1,) * nz, ny, nx)
+        if nrgb > 1:
+            chunks = chunks + (nrgb,)
+        arr = da.map_blocks(
+            self._dask_chunk,
+            chunks=chunks,
+            dtype=self.core_meta.dtype,
+        )
+        return _DaskArrayProxy(arr, self)
+
+    @property
+    def is_open(self) -> bool:
+        """Whether the underlying file is currently open"""
+        return bool(self._r.getCurrentFile())
+
+    @property
+    def filename(self) -> str:
+        """Return name of file handle."""
+        # return self._r.getCurrentFile()
+        return self._path
+
+    @property
+    def ome_xml(self) -> str:
+        """return OME XML string."""
+        with self:
+            store = self._r.getMetadataStore()
+
+            return str(store.dumpXML()) if store else ""
+
+    @property
+    def ome_metadata(self) -> "OME":
+        """Return OME object parsed by ome_types."""
+        import ome_types
```

Comment:
is this stuff expensive to just import at the top of the file?

Reply:
nope, just a habit i suppose.  particularly that one which is used elsewhere in this repo  (and therefore almost guaranteed to be imported already).  will move them